src/ObjCRuntime/Class.cs

```diff
@@ -42,6 +43,8 @@ public partial class Class : INativeObject
 		// We use the last significant bit of the IntPtr to store if this is a custom class or not.
 #pragma warning disable CS8618 // "Non-nullable field must contain a non-null value when exiting constructor." - we ensure these fields are non-null in other ways
 		static Dictionary<Type, IntPtr> type_to_class; // accessed from multiple threads, locking required.
+		static ConditionalWeakTable<Assembly, string> assembly_to_name; // accessed from multiple threads, but ConditionalWeakTables are thread-safe, so no locking required.
+		static Dictionary<ulong, MemberInfo?> token_to_member; // accessed from multiple threads, locking required.
```

Comment:
Does MemberInfo? really need to be nullable?  don't see a place in which is added as null. And all accesses are using TryGetValue.

Reply:
Yes, it needs to be nullable, because we add the return value from `ResolveFullTokenReference`, which returns a nullable value:

https://github.com/xamarin/xamarin-macios/blob/0e41cad6651eb4adad523f0f418d0f77716ebe95/src/ObjCRuntime/Class.cs#L418